prometheus/remote.proto

```diff
@@ -0,0 +1,70 @@
+// This file is copied (except for package name) from https://github.com/prometheus/prometheus/blob/master/storage/remote/remote.proto
+
+// Copyright 2016 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package prometheus; // change package from remote to prometheus
```

Comment:
Can we put this file in an internal package named remote? I do not want us doing any manual modifications to this file if at all possible. Something like this should work:

    prometheus/
      internal/
        remote/
          remote.proto

And then modify the `go generate` command to output the `remote.pb.go` file to this internal directory. Since it's in the remote folder, the package name will still be `remote` and it will be accessible to the `prometheus` package, but it won't be accessible to anyone else. That will make updating that file easier when we need to because there are no modifications.

Reply:
Why the `internal` before? Why not just put it in `remote`?

Reply:
It means it's not an interface we expose. While not necessary, I think it would be a good habit for us to start getting into.

Reply:
That feels odd to me. Go already has the public vs. private in a package. If we go down this route we'll end up having `internal` in every package in the project, no?